src/utils/service/service.js

```diff
@@ -149,31 +150,75 @@ class Service {
    *
    * @method get
    * @param {Number} id - the ID of the resource.
-   * @param {Function} onSuccess - a callback to trigger on success
-   * @param {Function} onError - a callback to trigger on error
+   * @param {Object} options - an object with options for query params and success/error callbacks
+   * @param {Function} _onError - [Deprecated] a callback to trigger on error
    * @return {Void}
    */
-  get = (id, onSuccess, onError) => {
-    this.client.get(String(id)).then(
-      this.handleResponse.bind(this, onSuccess),
-      this.handleResponse.bind(this, onError)
-    );
+
+  get = (id, options, _onError) => {
+    let request;
+
+    if (typeof (options) === 'function') {
+      Logger.deprecate('Passing onSuccess and onError as seperate arguments is deprecated. Please use `get(1, { onSuccess: function, onError: function, params { k: "v" } })`'); // eslint-disable-line max-len
+
+      request = this.client.get(String(id));
+
+      request.then(
+        this.handleResponse.bind(this, options),
+        this.handleResponse.bind(this, _onError)
+      );
+    } else {
+      if (typeof (options.params) === 'undefined') {
```

Comment:
could the else and if be collapsed into an `elseif`

Reply:
Not really. The `else` here is dealing with `options` not being a function while we deprecate `onSuccess` argument. The `if` on the next line is just dealing with `params` not being sent in the object.

Worth noting here this whole `else`/`if` thing is purely because we needed this to be backwards compatible, but needed to be able to have a way of sending in `params` to `axios`. Without defining a separate method entirely, this was the best way we could think of.

Reply:
Right, didn't look closely enough at the brackets. It's easier to see outside the diff view. 👍